state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -626,14 +639,31 @@ pub mod pallet {
 		DepositBoosted {
 			deposit_address: TargetChainAccount<T, I>,
 			asset: TargetChainAsset<T, I>,
-			amount: TargetChainAmount<T, I>,
+			amounts: BTreeMap<BoostPoolTier, TargetChainAmount<T, I>>,
 			deposit_details: <T::TargetChain as Chain>::DepositDetails,
+			channel_id: ChannelId,
 			// Ingress fee in the deposit asset. i.e. *NOT* the gas asset, if the deposit asset is
 			// a non-gas asset.
 			ingress_fee: TargetChainAmount<T, I>,
 			boost_fee: TargetChainAmount<T, I>,
 			action: DepositAction<T::AccountId>,
 		},
+		BoostFundsAdded {
+			account_id: T::AccountId,
+			boost_pool: BoostPoolId<T::TargetChain>,
+			amount: TargetChainAmount<T, I>,
+		},
+		AccountStoppedBoosting {
+			account_id: T::AccountId,
+			boost_pool: BoostPoolId<T::TargetChain>,
+			account_available_amount: TargetChainAmount<T, I>,
+			account_in_use_amounts: BTreeMap<BoostId, TargetChainAmount<T, I>>,
```

Comment:
Is it possible to have a different key for this map? 
Preferably something that we could use for identifying the channel related to these boosts. `ChannelId` would allow us to do that although it might not be the most ergonomic approach.

Thinking about it a little bit more, we could also just include an array of all boost ids this account contributed to instead of a map with the amounts. We will then need to expose a `boost_id: BoostId` inside `DepositBoosted` & `DepositFinalised.action.BoostersCredited` which will allow us to keep track of these `pending` amounts on our end.
Of course, that is, if the `BoostId` stays the same for a specific deposit throughout its lifecycle (`DepositBoosted -> DepositFinalised.action.BoostersCredited`).

This might be a cleaner approach as we can then also omit the `channel_id` property from this event.

What do you think @kylezs

Reply:
>  If so, will this still hold true in the case of a re-org scenario?

Yes. the ID is a counter that will only ever go up that is given when the deposit if prewitnessed.

> This might be a cleaner approach

I agree.